backend/runner/taskrun/data_update_executor.go

```diff
@@ -117,38 +117,49 @@ func (exec *DataUpdateExecutor) backupData(
 		return err
 	}
 
-	suffix := time.Now().Format("20060102150405")
-	selectIntoStatement, targetTableName := updateToSelect(statement, backupDatabaseName, suffix, issue.UID)
-	if _, err := driver.Execute(driverCtx, selectIntoStatement, db.ExecuteOptions{}); err != nil {
-		return err
-	}
-	createActivityPayload := api.ActivityPipelineTaskPriorBackupPayload{
-		TaskID: task.ID,
-		BackupSchemaMetadata: []api.SchemaMetadata{
-			{
-				Table: targetTableName,
-			},
-		},
-		IssueName: issue.Title,
-		TaskName:  task.Name,
-	}
-	bytes, err := json.Marshal(createActivityPayload)
+	suffix := "_" + time.Now().Format("20060102150405")
+	statements, err := base.TransformDMLToSelect(instance.Engine, statement, database.DatabaseName, backupDatabaseName, suffix)
 	if err != nil {
-		return errors.Wrapf(err, "failed to marshal ActivityIssueCreate activity")
-	}
-	activityCreate := &store.ActivityMessage{
-		CreatorUID:   api.SystemBotID,
-		ContainerUID: issue.UID,
-		Type:         api.ActivityPipelineTaskPriorBackup,
-		Level:        api.ActivityInfo,
-		Payload:      string(bytes),
-	}
-	if _, err := exec.activityManager.CreateActivity(ctx, activityCreate, &activity.Metadata{Issue: issue}); err != nil {
-		slog.Error("failed to create activity",
-			slog.Int("task", task.ID),
-			log.BBError(err),
-		)
+		return errors.Wrap(err, "failed to transform DML to select")
+	}
+
+	for _, statement := range statements {
+		if _, err := driver.Execute(driverCtx, statement.Statement, db.ExecuteOptions{}); err != nil {
```

Comment:
Should we use a single transaction for all transformed select statement?

Reply:
I don't think it's a good idea. It will make a so-big transaction and hand locks a long time.